universityofadelaide/shepherd
# Feature/webops 311 deleting dc leaves orphans

Issue : 
Current used versions of OpenShift 3.7+ have an issue where an deleted environment leaves remaining pods and replication controllers. Previously this did not happen and the restart policy hadn't changed. 

Looking into the issue the agreed best practice approach is to scale down the replicas to zero - nullifying the restart policy before triggering the removal of deployment config, replication controllers etc. Now this doesn't answer exactly why the issue was occurring, but seems to me that replication controller was firing the restart policy before the deploymentConfig was entirely deleted - the pods will have been sent a signal to delete causing the rc to react as it should to the policy. End result was orphaned pods.

How to test : 
Checkout this branch, create and environment and delete it. There should be no remaining rc's or pods. 

Also see and checkout https://github.com/universityofadelaide/openshift-client/pull/24
